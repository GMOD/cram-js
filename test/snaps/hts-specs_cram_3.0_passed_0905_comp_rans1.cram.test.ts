import { testFile } from '../testUtil'
testFile('hts-specs/cram/3.0/passed/0905_comp_rans1.cram')
