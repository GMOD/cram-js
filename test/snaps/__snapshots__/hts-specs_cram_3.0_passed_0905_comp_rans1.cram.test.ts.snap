// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole hts-specs/cram/3.0/passed/0905_comp_rans1.cram without error 1`] = `
[
  {
    "fileId": "_.cram",
    "magic": "CRAM",
    "majorVersion": 3,
    "minorVersion": 0,
  },
  {
    "containerHeader": {
      "_endPosition": 45,
      "_size": 19,
      "alignmentSpan": 0,
      "crc32": 3124391203,
      "landmarks": [
        0,
        185,
      ],
      "length": 286,
      "numBases": 0,
      "numBlocks": 2,
      "numLandmarks": 2,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 45,
        "_size": 19,
        "alignmentSpan": 0,
        "crc32": 3124391203,
        "landmarks": [
          0,
          185,
        ],
        "length": 286,
        "numBases": 0,
        "numBlocks": 2,
        "numLandmarks": 2,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/nfs/users/nfs_j/jkb/work/samtools_master/hts-specs/test/CRAM/passed/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "ID",
                "value": "rg",
              },
              {
                "tag": "SM",
                "value": "test",
              },
            ],
            "tag": "RG",
          },
          {
            "data": [
              {
                "tag": "ID",
                "value": "rg2",
              },
              {
                "tag": "SM",
                "value": "test",
              },
            ],
            "tag": "RG",
          },
        ],
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "CHROMOSOME_I",
              },
              {
                "tag": "LN",
                "value": "1009800",
              },
              {
                "tag": "M5",
                "value": "8ede36131e0dbf3417807e48f77f3ebd",
              },
              {
                "tag": "UR",
                "value": "/nfs/users/nfs_j/jkb/work/samtools_master/hts-specs/test/CRAM/passed/ce.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "ID",
                "value": "rg",
              },
              {
                "tag": "SM",
                "value": "test",
              },
            ],
            "tag": "RG",
          },
          {
            "data": [
              {
                "tag": "ID",
                "value": "rg2",
              },
              {
                "tag": "SM",
                "value": "test",
              },
            ],
            "tag": "RG",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": false,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 6,
          "parameters": {
            "length": 11,
            "offset": 0,
          },
          "parametersBytes": 2,
        },
        "BB": {
          "codecId": 4,
          "parameters": {
            "lengthsEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 42,
              },
              "parametersBytes": 1,
            },
            "valuesEncoding": {
              "codecId": 1,
              "parameters": {
                "blockContentId": 37,
              },
              "parametersBytes": 1,
            },
          },
          "parametersBytes": 6,
        },
        "BF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 15,
          },
          "parametersBytes": 1,
        },
        "BS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 31,
          },
          "parametersBytes": 1,
        },
        "CF": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 16,
          },
          "parametersBytes": 1,
        },
        "FC": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              88,
            ],
          },
          "parametersBytes": 4,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              3,
            ],
          },
          "parametersBytes": 4,
        },
        "FP": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 28,
          },
          "parametersBytes": 1,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              40,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 18,
          },
          "parametersBytes": 1,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              100,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 352,
      "_size": 21,
      "alignmentSpan": 300,
      "crc32": 4249423309,
      "landmarks": [
        154,
      ],
      "length": 729,
      "numBases": 400,
      "numBlocks": 10,
      "numLandmarks": 1,
      "numRecords": 4,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1000,
    },
    "data": [
      {
        "_endPosition": 352,
        "_size": 21,
        "alignmentSpan": 300,
        "crc32": 4249423309,
        "landmarks": [
          154,
        ],
        "length": 729,
        "numBases": 400,
        "numBlocks": 10,
        "numLandmarks": 1,
        "numRecords": 4,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1000,
      },
      {
        "features": [
          {
            "alignmentStart": 1000,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1200,
              "readName": "r1",
              "sequenceId": 0,
              "uniqueId": 513,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 1,
                "pos": 51,
                "refPos": 1050,
              },
              {
                "code": "X",
                "data": 3,
                "pos": 52,
                "refPos": 1051,
              },
              {
                "code": "X",
                "data": 2,
                "pos": 53,
                "refPos": 1052,
              },
            ],
            "readGroupId": 0,
            "readLength": 100,
            "readName": "r1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 512,
          },
          {
            "alignmentStart": 1200,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1000,
              "readName": "r1",
              "sequenceId": 0,
              "uniqueId": 512,
            },
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 0,
                "pos": 51,
                "refPos": 1250,
              },
              {
                "code": "X",
                "data": 3,
                "pos": 52,
                "refPos": 1251,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 53,
                "refPos": 1252,
              },
            ],
            "readGroupId": 0,
            "readLength": 100,
            "readName": "r1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 513,
          },
          {
            "alignmentStart": 1000,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1200,
              "readName": "r2",
              "sequenceId": 0,
              "uniqueId": 515,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 1,
                "pos": 51,
                "refPos": 1050,
              },
              {
                "code": "X",
                "data": 3,
                "pos": 52,
                "refPos": 1051,
              },
              {
                "code": "X",
                "data": 2,
                "pos": 53,
                "refPos": 1052,
              },
            ],
            "readGroupId": 1,
            "readLength": 100,
            "readName": "r2",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 514,
          },
          {
            "alignmentStart": 1200,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 100,
            "mappingQuality": 40,
            "mate": {
              "alignmentStart": 1000,
              "readName": "r2",
              "sequenceId": 0,
              "uniqueId": 514,
            },
            "qualityScores": [
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              2,
              31,
              33,
              30,
              23,
              33,
              30,
              33,
              32,
              31,
              31,
              35,
              35,
              33,
              34,
              35,
              35,
              34,
              33,
              34,
              31,
              34,
              35,
              34,
              35,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
              34,
            ],
            "readBases": undefined,
            "readFeatures": [
              {
                "code": "X",
                "data": 0,
                "pos": 51,
                "refPos": 1250,
              },
              {
                "code": "X",
                "data": 3,
                "pos": 52,
                "refPos": 1251,
              },
              {
                "code": "X",
                "data": 1,
                "pos": 53,
                "refPos": 1252,
              },
            ],
            "readGroupId": 1,
            "readLength": 100,
            "readName": "r2",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 300,
            "templateSize": undefined,
            "uniqueId": 515,
          },
        ],
        "header": {
          "_endPosition": 552,
          "_size": 41,
          "compressedSize": 37,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            131,
            232,
            129,
            44,
            4,
            0,
            8,
            7,
            11,
            12,
            15,
            16,
            18,
            28,
            31,
            255,
            255,
            255,
            255,
            15,
            188,
            14,
            187,
            152,
            12,
            130,
            56,
            146,
            25,
            54,
            223,
            160,
            201,
            234,
            161,
            96,
          ],
          "contentId": 0,
          "contentPosition": 511,
          "contentType": "MAPPED_SLICE_HEADER",
          "crc32": 2440517271,
          "parsedContent": {
            "_endPosition": 389,
            "_size": 37,
            "contentIds": [
              11,
              12,
              15,
              16,
              18,
              28,
              31,
            ],
            "md5": [
              188,
              14,
              187,
              152,
              12,
              130,
              56,
              146,
              25,
              54,
              223,
              160,
              201,
              234,
              161,
              96,
            ],
            "numBlocks": 8,
            "numContentIds": 7,
            "numRecords": 4,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 300,
            "refSeqStart": 1000,
          },
          "uncompressedSize": 37,
        },
      },
    ],
  },
]
`;
